base_layer/wallet/src/transaction_service/multisig/session.rs

```diff
@@ -0,0 +1,545 @@
+        let fee_per_gram = MicroMinotari::from(1);
+        let payment_id =
+            MemoField::new_address_and_data(recipient.clone(), fee_per_gram, true, TxType::PaymentToOther, user_data)
+                .map_err(|e| TransactionServiceError::Other(format!("Failed to create MemoField: {}", e)))?;
+
+        let mut selected_criteria = UtxoSelectionCriteria::default();
+
+        selected_criteria.excluding_multisig = true;
+
+        let mut stp = output_manager
+            .prepare_transaction_to_send(
+                tx_id,
+                amount,
+                selected_criteria,
+                OutputFeatures::default(),
+                fee_per_gram,
+                push_pubkey_script(&Default::default()),
+                Covenant::default(),
+            )
+            .await?;
+
+        let sender_offset_key = self
+            .resources
+            .transaction_key_manager_service
+            .get_next_key(TransactionKeyManagerBranch::OneSidedSenderOffset.get_branch_key())
+            .await?;
+
+        let recipient_spend_key = recipient.public_spend_key();
+
+        let sender_offset_public_key = self
+            .resources
+            .transaction_key_manager_service
+            .get_public_key_at_key_id(&sender_offset_key.key_id)
+            .await?;
+
+        let encrypted_data_shared_secret = key_manager
+            .get_diffie_hellman_shared_secret(
+                &sender_offset_key.key_id,
+                recipient
+                    .public_view_key()
+                    .ok_or(TransactionServiceError::Other("Missing public view key".to_string()))?,
+            )
+            .await?;
+
+        let encryption_private_key = shared_secret_to_output_encryption_key(&encrypted_data_shared_secret)
+            .map_err(|e| TransactionServiceError::Other(format!("Failed to derive encryption key: {}", e)))?;
+
+        let encryption_key_id = key_manager.import_key(encryption_private_key.clone()).await?;
+
+        let mut ephemeral_pubkeys = Vec::new();
+        for pk in &public_keys {
+            // 1. Get the shared secret (aD)
+            let shared_secret = key_manager
+                .get_diffie_hellman_shared_secret(&sender_offset_key.key_id, &pk)
+                .await?;
+
+            // 2. Hash the shared secret for stealth domain separation
+            let stealth_hash = diffie_hellman_stealth_domain_hasher(shared_secret);
+
+            let mask_key_id = &key_manager
+                .import_key(PrivateKey::from_uniform_bytes(stealth_hash.as_ref())?)
+                .await?;
+
+            let ephemeral_pubkey = key_manager
+                .clone()
+                .stealth_address_script_spending_key(&mask_key_id, pk)
+                .await?
+                .to_public_key()?;
+
+            ephemeral_pubkeys.push(CompressedPublicKey::new_from_pk(ephemeral_pubkey.clone()));
+        }
+
+        if m == 0 || (m as usize) > ephemeral_pubkeys.len() {
```

Comment:
```suggestion
        if m == 0 || (m as usize) > public_keys.len() {
```
And then this can be infront, you dont have to do all the crypto for this check